geode-core/src/main/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderEventProcessor.java

```diff
@@ -78,13 +88,18 @@ protected void initializeMessageQueue(String id, boolean cleanQueues) {
     }
 
     ParallelGatewaySenderQueue queue =
-        new ParallelGatewaySenderQueue(sender, targetRs, index, nDispatcher, cleanQueues);
+        new ParallelGatewaySenderQueue(sender, targetRs, index, nDispatcher, cleanQueues,
+            recoverQueuesOnly);
 
-    queue.start();
+    if (!recoverQueuesOnly) {
+      queue.start();
+    }
     this.queue = queue;
 
-    if (queue.localSize() > 0) {
-      queue.notifyEventProcessorIfRequired();
+    if (!recoverQueuesOnly) {
+      if (queue.localSize() > 0) {
+        queue.notifyEventProcessorIfRequired();
+      }
```

Comment:
Try to avoid nesting especially when it's unnecessary.
```
if (!recoverQueuesOnly && queue.localSize() > 0) {
  queue.notifyEventProcessorIfRequired();
}
```
Or you could extract that conditional to a small private method with a descriptive name to make it easier to understand. It's hard to figure out what `!recoverQueuesOnly && queue.localSize() > 0` really means unless I study the rest of the code in this class.